evm_loader/program/src/executor/state.rs

```diff
@@ -284,18 +285,23 @@ impl<'a, B: AccountStorage> Database for ExecutorState<'a, B> {
         Ok(self.backend.code_size(from_address).await)
     }
 
-    async fn code_hash(&self, from_address: Address, chain_id: u64) -> Result<[u8; 32]> {
-        use solana_program::keccak::hash;
+    async fn code_hash(&self, address: Address, chain_id: u64) -> Result<[u8; 32]> {
```

Comment:
As @anton-lisanin mentioned [here](https://github.com/neonlabsorg/neon-evm/pull/253#issuecomment-1887165009), I suggest inlining `code_hash` method into `opcode_extcodehash` method, since it is only used there and decouples `Database` trait from `code_hash` logic, which is not its concern.

Reply:
Putting the code inside of the opcode handler felt like mixing different levels of abstraction. Instead I chose to pull the implementation out into an extension trait `DatabaseExt` which contains functions that can be implemented on top of anything that implements `Database`. Alternatively, we could also make use of default function implementations, free standing functions, or just inlining everything as initially suggested.

Reply:
I think it is a bit of over abstraction, but I am ok with it too.